x-pack/plugins/security_solution/public/common/links/links.ts

```diff
@@ -179,6 +182,20 @@ const processAppLinks = (appLinks: AppLinkItems, linksPermissions: LinksPermissi
     return acc;
   }, []);
 
+export const isLinkUiSettingsAllowed = (link: LinkItem, { uiSettingsClient }: LinksPermissions) => {
+  if (link.uiSettingRequired) {
```

Comment:
can we invert this check and short circuit it so we don't have nested conditionals?

Reply:
Yes, but it required an unreachable return statement:
```js
export const isLinkUiSettingsAllowed = (link: LinkItem, { uiSettingsClient }: LinksPermissions) => {
  if (!link.uiSettingRequired) {
    return true;
  }

  if (typeof link.uiSettingRequired === 'string') {
    return uiSettingsClient.get(link.uiSettingRequired) === true;
  }

  if (typeof link.uiSettingRequired === 'object') {
    return uiSettingsClient.get(link.uiSettingRequired.key) === link.uiSettingRequired.value;
  }

  // this return statement is required by TS
  return false;
};
```